packages/rum-core/src/domain/parentContexts.ts

```diff
@@ -37,25 +37,29 @@ export function startParentContexts(lifeCycle: LifeCycle, session: RumSession):
   let previousActions: Array<PreviousContext<ActionContext>> = []
 
   lifeCycle.subscribe(LifeCycleEventType.VIEW_CREATED, (currentContext) => {
-    if (currentView) {
-      previousViews.unshift({
-        context: buildCurrentViewContext(),
-        endTime: currentContext.startClocks.relative,
-        startTime: currentView.startClocks.relative,
-      })
-    }
     currentView = currentContext
     currentSessionId = session.getId()
   })
 
   lifeCycle.subscribe(LifeCycleEventType.VIEW_UPDATED, (currentContext) => {
     // A view can be updated after its end.  We have to ensure that the view being updated is the
     // most recently created.
-    if (currentView!.id === currentContext.id) {
+    if (currentView && currentView.id === currentContext.id) {
       currentView = currentContext
     }
   })
 
+  lifeCycle.subscribe(LifeCycleEventType.VIEW_ENDED, ({ id, endClocks }) => {
+    if (currentView && currentView.id === id) {
```

Comment:
Do we need to check the view id? If so, I think we should do the same in the recorder.

Reply:
indeed, we should not